src/js/components/Box.js

```diff
@@ -3,11 +3,20 @@
 import React, { Component, PropTypes } from 'react';
 import KeyboardAccelerators from '../utils/KeyboardAccelerators';
 import Intl from '../utils/Intl';
+import Props from '../utils/Props';
 import SkipLinkAnchor from './SkipLinkAnchor';
 
 const CLASS_ROOT = "box";
 
 export default class Box extends Component {
+  constructor(props) {
+    super(props);
+    this.restProps = Props.omit(props, Object.keys(Box.propTypes));
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.restProps = Props.omit(nextProps, Object.keys(Box.propTypes));
```

Comment:
Instead of duplicating code, better to move it back to `render`, which will be the same (sorry about my misleading comment above). Checking props change would probably be better to have in `shouldComponentUpdate`, but I don't think we should solve this here, though adding a separate PR with performance optimizations would be great.